### PR TITLE
feat: support using keyword

### PR DESCRIPTION
**Pull request by ffMathy**
Today, if a test has the `using` keyword in it (a new TypeScript feature), Vitest won't run or discover that particular test. This fixes that problem.

### CODE REVIEW

**Comment by MilanKovacic** [comment]
Can we add the "using" keyword to one of the "samples" test cases?

File: src/pure/parsers/helper.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ const commonPlugins: ParserPlugin[] = [
   'optionalChaining',
   'partialApplication',
   'throwExpressions',
+  'explicitResourceManagement',
```
Comment (MilanKovacic) [inline src/pure/parsers/helper.ts:32]: Can we add a comment specifying that this relates to "using" keyword?

Comment (ffMathy) [reply src/pure/parsers/helper.ts]: Sure! Done! 👍

**Comment by ffMathy** [comment]
> Can we add the "using" keyword to one of the "samples" test cases?

Done! I also added an actual unit test to the parser which verifies that it is working.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**